core/src/main/scala/chisel3/Aggregate.scala

```diff
@@ -182,10 +183,15 @@ trait VecFactory extends SourceInfoDoc {
     implicit sourceInfo: SourceInfo
   ): UInt = {
     val w = (n - 1).bitLength
-    if (n <= 1) 0.U
-    else if (idx.width.known && idx.width.get <= w) idx
-    else if (idx.width.known) idx(w - 1, 0)
-    else (idx | 0.U(w.W))(w - 1, 0)
+    val name = "_vec_index_truncate"
+    // Give any truncation logic a name to make debugging easier
+    autoNameRecursively(name)(prefix(name) {
+      if (n <= 1) WireInit(0.U) // Need the Wire otherwise we emit vec[0] which is illegal FIRRTL.
```

Comment:
Is it cleaner to use a zero-width, invalidated wire here?

Reply:
Great question, I'll try it out. This wire is closer to what Chisel had been emitting, but that might be cleaner and not rely on something that probably should be illegal in FIRRTL anyway.

Reply:
I tried it, it seemed to work for the most part but caused some subtle failure in the QueueSpec integration tests. I'm not convinced it's a real issue, but I couldn't quickly figure out what's wrong so I'll just leave it alone for now. We want to get rid of this truncation anyway so I'm not _that_ concerned with making sure it's the most elegant.